build.sbt

```diff
@@ -138,116 +141,91 @@ lazy val `root3` =
 
 lazy val zioConfig = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .in(file("core"))
-  .settings(stdSettings("zio-config"))
-  .settings(crossProjectSettings)
-  .enablePlugins(BuildInfoPlugin)
-  .settings(buildInfoSettings("zio.config"))
+  .settings(
+    stdSettings(
+      name = "zio-config",
+      packageName = Some("zio.config"),
+      enableCrossProject = true
+    )
+  )
   .settings(macroDefinitionSettings)
+  .settings(enableZIO())
   .settings(
-    crossScalaVersions --= Seq("2.11"),
     libraryDependencies ++= Seq(
-      "dev.zio"                %% "zio"                     % zioVersion,
-      "org.scala-lang.modules" %% "scala-collection-compat" % "2.8.1",
-      "dev.zio"                %% "zio-test"                % zioVersion % Test
-    ),
-    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+      "org.scala-lang.modules" %% "scala-collection-compat" % "2.8.1"
+    )
   )
 
-lazy val zioConfigJS     = zioConfig.js
-  .settings(libraryDependencies += "dev.zio" %%% "zio-test-sbt" % zioVersion % Test)
-lazy val zioConfigJVM    = zioConfig.jvm
-  .settings(dottySettings)
-  .settings(libraryDependencies += "dev.zio" %%% "zio-test-sbt" % zioVersion % Test)
-lazy val zioConfigNative = zioConfig.native
-  .settings(nativeSettings)
+lazy val zioConfigJS     = zioConfig.js.settings(crossScalaVersions -= scala211.value)
+lazy val zioConfigJVM    = zioConfig.jvm.settings(scala3Settings)
+lazy val zioConfigNative = zioConfig.native.settings(nativeSettings)
 
-lazy val zioConfigAws    = crossProject(JVMPlatform)
+lazy val zioConfigAws = crossProject(JVMPlatform)
   .in(file("aws"))
-  .settings(stdSettings("zio-config-aws"))
-  .settings(crossProjectSettings)
-  .settings(dottySettings)
+  .settings(stdSettings("zio-config-aws", enableCrossProject = true))
+  .settings(enableZIO(enableStreaming = true))
+  .settings(scala3Settings)
   .settings(
-    crossScalaVersions --= Seq("2.11"),
     libraryDependencies ++= Seq(
-      "com.amazonaws" % "aws-java-sdk-ssm" % awsVersion,
-      "dev.zio"      %% "zio-streams"      % zioVersion,
-      "dev.zio"      %% "zio-test"         % zioVersion % Test,
-      "dev.zio"      %% "zio-test-sbt"     % zioVersion % Test
-    ),
-    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+      "com.amazonaws" % "aws-java-sdk-ssm" % awsVersion
+    )
```

Comment:
Much better for any new contributor, is to be able to say

```scala
lazy val zioConfigAws = jvmModule("was", "zio-config-aws').withDependencies().dependOn(zioConfig)
```

As mentioned above withDependencies can take care of weird cases, such as excluding if Scala-version is `x` etc.

Reply:
I agree, will work on this!